cads_adaptors/adaptors/multi.py

```diff
@@ -157,16 +200,11 @@ def pre_mapping_modifications(self, request: dict[str, Any]) -> dict[str, Any]:
 
     def retrieve_list_of_results(self, request: Request) -> list[str]:
         request = self.normalise_request(request)
```

Comment:
This could be a tidy solution for not remerging the mapped request.

If set to true, we intersect_constraints in the retrieve-api which calls normalise_request directly to check for valid requests.

When running the request, we do not intersect constraints at the top level, but pass the intersect_constraints value to the sub-adaptors, which will do the intersection with their portion of the sub-request.

```suggestion
        self.intersect_constraints_bool = False
        request = self.normalise_request(request)
```

Reply:
Then we should merge only if there is more than one mapped_requests to be merged (just in case at some point in the future that would become possible and/or we missed a split that happens before).

Reply:
Agree with your correction on the previous point, because when we call normalise_request from the retrieve-api for validating requests, we can to do the intersection on the parent class.